executor/executor.go

```diff
@@ -1744,7 +1744,13 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	case *ast.LoadDataStmt:
 		sc.DupKeyAsWarning = true
 		sc.BadNullAsWarning = true
-		sc.TruncateAsWarning = !vars.StrictSQLMode
+		// With IGNORE or LOCAL, data-interpretation errors become warnings and the load operation continues,
+		// even if the SQL mode is restrictive. For examples, see Column Value Assignment.
+		if stmt.IsLocal || stmt.OnDuplicate == ast.OnDuplicateKeyHandlingIgnore {
+			sc.TruncateAsWarning = true
+		} else {
+			sc.TruncateAsWarning = !vars.StrictSQLMode
```

Comment:
In what case can TiDB reach this line？

Reply:
eg: load data without ignore keyword and local keyword

Reply:
I think `load data` without `local` is not supported in TiDB.

Reply:
yes

Reply:
@jackysp means that `sc.TruncateAsWarning = true` , no if.

Reply:
sure